### PR TITLE
Add Localization Support for SPM Integrations

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

 - Add support for localization with Swift Package Manager:
     - Move localized strings under a "Resources" directory (per [Apple's requirements](https://developer.apple.com/documentation/swift_packages/localizing_package_resources))
     - Update logic to get the bundle with the localized strings (see [this article](https://darjeelingsteve.com/articles/How-to-Use-Module-Resources-in-Objective-C-SPM-Packages.html))
 - Update the SPMTest app to include a simple UI test that checks that localizations are working
 - Update the `build` GitHub Actions workflow to run that UI test

I think we should add a similar UI test for CocoaPods and perhaps our demo app, but we can do that in a separate PR.

 ### Checklist

 - ~Added a changelog entry~ (this falls under the "Add support for SPM" entry)

- @sestevens

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,9 +19,13 @@ jobs:
         uses: actions/checkout@v2
       - name: Use Xcode 12.4
         run: sudo xcode-select -switch /Applications/Xcode_12.4.app
+      - name: Install xcpretty
```
Comment (scannillo) [inline .github/workflows/build.yml:22]: Random - when I was trialing testflight stuff via GH Actions, I didn't need to install `xcpretty` to use it. 🤷‍♀️

File: Sources/BraintreeUIKit/Localization/BTUIKLocalizedString.m [hunk 2]
```diff
@@ -19,15 +19,18 @@ + (NSBundle *)localizationBundle {
...
-
-    // CocoaPods creates "Braintree-UIKit-Localization.bundle", so we check for that first.
-    // If not found, use the BraintreeUIKit bundle, which is where Carthage will find the localized strings.
-    return localizationBundlePath ? [NSBundle bundleWithPath:localizationBundlePath] : [NSBundle bundleForClass:self.class];
+    return [NSBundle bundleWithPath:localizationBundlePath];
+#elif SWIFT_PACKAGE
+    return SWIFTPM_MODULE_BUNDLE;
+#else
+    return [NSBundle bundleForClass:self.class];
+#endif
```
Comment (scannillo) [inline Sources/BraintreeUIKit/Localization/BTUIKLocalizedString.m:33]: Nice cleanup here. Also nice job researching this!

**Review by scannillo — APPROVED** [review]
🏁

File: .github/workflows/build.yml (hunk 1 above)
Comment (sestevens) [reply .github/workflows/build.yml:22]: Oooh, good to know! I'll try removing it.

File: Sources/BraintreeUIKit/Localization/BTUIKLocalizedString.m (hunk 2 above)
Comment (sestevens) [reply Sources/BraintreeUIKit/Localization/BTUIKLocalizedString.m:33]: Thank you! 😊

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -19,9 +19,13 @@ jobs:
...
+        run: gem install xcpretty
```
Comment (sestevens) [inline .github/workflows/build.yml:23]:
```suggestion
```

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**